pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -46,55 +47,32 @@ const (
 	sigImageVersion = "202505.27.0"
 )
 
-func getExpectedTestCIGImages(imageFamily string, fipsMode *v1beta1.FIPSMode, version string, kubernetesVersion string) []imagefamily.NodeImage {
-	var images []imagefamilytypes.DefaultImageOutput
-	if imageFamily == v1beta1.Ubuntu2204ImageFamily {
-		images = imagefamily.Ubuntu2204{}.DefaultImages(false, fipsMode)
-	} else if imageFamily == v1beta1.AzureLinuxImageFamily {
-		if imagefamily.UseAzureLinux3(kubernetesVersion) {
-			images = imagefamily.AzureLinux3{}.DefaultImages(false, fipsMode)
-		} else {
-			images = imagefamily.AzureLinux{}.DefaultImages(false, fipsMode)
-		}
+func renderExpectedCIGNodeImages(
+	fam imagefamily.ImageFamily,
+	fips *v1beta1.FIPSMode,
+	version string,
+) []imagefamily.NodeImage {
+	defaultImages := fam.DefaultImages(false, fips)
+	out := make([]imagefamily.NodeImage, 0, len(defaultImages))
+	for _, img := range defaultImages {
+		id := imagefamily.BuildImageIDCIG(img.PublicGalleryURL, img.ImageDefinition, version)
+		out = append(out, imagefamily.NodeImage{ID: id, Requirements: img.Requirements})
 	}
-	nodeImages := []imagefamily.NodeImage{}
-	for _, image := range images {
-		nodeImages = append(nodeImages, imagefamily.NodeImage{
-			ID:           fmt.Sprintf("/CommunityGalleries/%s/images/%s/versions/%s", image.PublicGalleryURL, image.ImageDefinition, version),
-			Requirements: image.Requirements,
-		})
-	}
-	return nodeImages
+	return out
 }
 
-//nolint:unparam // might always be using the same version in test, but could change in the future
-func getExpectedTestSIGImages(imageFamily string, fipsMode *v1beta1.FIPSMode, version string, kubernetesVersion string) []imagefamily.NodeImage {
-	var images []imagefamilytypes.DefaultImageOutput
-	var actualImageFamily imagefamily.ImageFamily
-	if imageFamily == v1beta1.UbuntuImageFamily {
-		if lo.FromPtr(fipsMode) == v1beta1.FIPSModeFIPS {
-			actualImageFamily = &imagefamily.Ubuntu2004{}
-		} else {
-			actualImageFamily = &imagefamily.Ubuntu2204{}
-		}
-	} else if imageFamily == v1beta1.Ubuntu2204ImageFamily {
-		actualImageFamily = &imagefamily.Ubuntu2204{}
-	} else if imageFamily == v1beta1.AzureLinuxImageFamily {
-		if imagefamily.UseAzureLinux3(kubernetesVersion) {
-			actualImageFamily = &imagefamily.AzureLinux3{}
-		} else {
-			actualImageFamily = &imagefamily.AzureLinux{}
-		}
-	}
-	images = actualImageFamily.DefaultImages(true, fipsMode)
-	nodeImages := []imagefamily.NodeImage{}
-	for _, image := range images {
-		nodeImages = append(nodeImages, imagefamily.NodeImage{
-			ID:           fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/galleries/%s/images/%s/versions/%s", sigSubscription, image.GalleryResourceGroup, image.GalleryName, image.ImageDefinition, version),
-			Requirements: image.Requirements,
-		})
+func renderExpectedSIGNodeImages(
+	fam imagefamily.ImageFamily,
+	fips *v1beta1.FIPSMode,
+	version, sigSub string, //nolint:unparam
```

Comment:
I would just remove it, until it is actually needed. Again, the ergonomics / readability of tests themselves is what's primary ...

Reply:
will followup on this in followup pr for a couple of other cleanup things.